packages/json-file/src/JsonFile.ts

```diff
@@ -43,7 +44,7 @@ const DEFAULT_OPTIONS: Options<*> = {
  * and object with string keys and either objects or primitive types as values.
  * @type {[type]}
  */
-export default class JsonFile<JSONObject: JSONT> {
+export default class JsonFile<JSONObject extends JSONT> {
```

Comment:
In TS and other languages, the convention for generic type parameter names is to use one or two characters usually:
```suggestion
export default class JsonFile<J extends JSONT> {
```

(maybe it is not actually a good convention, but when reading through the rest of this file it makes it more evident which types are generics and which are concrete types)

Reply:
What do you think about prefixing generic types with `T`, for example `TJSONObject`? This makes it really easy to distinguish them from concrete types, yet allows us to use descriptive names.

Reply:
I think that's fine, too.